shell/client/grainview.js

```diff
@@ -251,41 +254,54 @@ GrainView.prototype.revealIdentity = function () {
       identity = _.findWhere(identities, {id: token.owner.user.identityId});
     }
   }
-  this._userIdentity.set(identity);
+  this._userIdentityId.set(identity.id);
   this._dep.changed();
 }
 
 GrainView.prototype.doNotRevealIdentity = function () {
-  this._userIdentity.set(false);
+  this._userIdentityId.set(false);
```

Comment:
As explained in the comments above: `false` means incognito; `undefined` means we still need to decide whether to reveal an identity.